snforge_std/src/errors.cairo

```diff
@@ -0,0 +1,31 @@
+use core::option::OptionTrait;
+use core::starknet::SyscallResult;
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
+pub enum PanicDataOrString {
+    String: ByteArray,
+    PanicData: Array<felt252>,
+}
+
+type ExtendedSyscallResult<T> = Result<T, PanicDataOrString>;
+
+trait SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T>;
+}
+
+impl SyscallResultStringErrorTraitImpl<T> of SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T> {
```

Comment:
Wondering if this shouldn't return be wrapped in yet another result 🤔

Reply:
Nvm, you convinced me above^

Reply:
I actually don't think it should be, if anything fails here it means we have a bug since we serialized the string ourselves